src/utils/truffle.ts

```diff
@@ -0,0 +1,126 @@
+import { Address, ChainURI, TransactionHash, ABI, Compiler } from 'ethpm/package';
+
+
+interface DeploymentData {
+  contract_type: string,
+  address: Address,
+  transaction: TransactionHash,
+}
+
+interface Network {
+  address: Address,
+  transactionHash: TransactionHash,
+}
+
+interface Artifact {
+  abi: ABI,
+  bytecode: any,
+  deployedBytecode: any,
+  contractName: string,
+  compiler: Compiler,
+  devdoc: any,
+  userdoc: any,
+  metadata: string,
+  networks: Record<string, Network>,
+}
+
+
+function parseBytecode(bytecode: string) {
```

Comment:
Also for this, you can probably start with this shim: https://github.com/trufflesuite/truffle/blob/develop/packages/workflow-compile/shims.js#L59-L86